src/main/java/edu/tamu/app/controller/UserController.java

```diff
@@ -27,25 +28,32 @@
 
     /**
      * Websocket endpoint to request credentials.
-     * 
+     *
      * @param credentials
      * @ApiCredentials Credentials
-     * 
+     *
      * @return ApiResponse
-     * 
+     *
      */
     @RequestMapping("/credentials")
     @PreAuthorize("hasRole('ANONYMOUS')")
     public ApiResponse credentials(@WeaverCredentials Credentials credentials) {
```

Comment:
This is not the ideal way we would handle this. The argument resolver for Credentials should never return null, but should throw an exception if it cannot extract credentials from the principal. The controller would then throw an exception and a controller advice would handle the exception and return an appropriate response to the client.

https://github.com/TAMULib/Weaver-Webservice-Core/blob/2.x/auth/src/main/java/edu/tamu/weaver/auth/resolver/WeaverCredentialsArgumentResolver.java#L21

This file should be updated to throw new NoCredentialsFoundException() instead of return null. It would require making said exception and then have a global @ControllerAdvice implement @ExceptionHandler(NoCredentialsFoundException.class)